### PR TITLE
feat: support color&shape selection in shape mode

### PR DESCRIPTION
**Pull request by himself65**

### CODE REVIEW

**Comment by Himself65** [comment]
```bash
> tsc && vite build

vite v3.1.1 building for production...
✓ 360 modules transformed.
dist/style.css   6.86 KiB / gzip: 1.56 KiB
dist/index.js    2395.60 KiB / gzip: 564.75 KiB

```

**Comment by Himself65** [comment]
```
> @blocksuite/blocks@0.3.0-alpha.7 build /home/himself65/Code/blocksuite/packages/blocks
> tsc && vite build

vite v3.1.1 building for production...
✓ 361 modules transformed.
dist/style.css   6.86 KiB / gzip: 1.56 KiB
dist/index.js    2415.36 KiB / gzip: 569.92 KiB
```

**Comment by Himself65** [comment]
Screenshot from 2022-12-13 16-46-18

**Comment by himself65** [comment]
```
> tsc && vite build

vite v3.2.5 building for production...
✓ 125 modules transformed.
dist/style.css   6.86 KiB / gzip: 1.56 KiB
dist/index.js    708.39 KiB / gzip: 165.92 KiB

Process finished with exit code 0

```

File: packages/blocks/src/__internal__/utils/shape.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  minY = 'minY',
+  midY = 'midY',
+  maxY = 'maxY',
+}
+
+export type Snap =
+  | { id: SnapPoints; isSnapped: false }
+  | {
+      id: SnapPoints;
+      isSnapped: true;
```
Comment (doodlewind) [inline packages/blocks/src/__internal__/utils/shape.ts:121]: FWIW it's possible that snapping may be enabled for group blocks. Not sure if we could make its data structure general-purpose. The current port looks good, though.

**Comment by himself65** [comment]
Screenshot from 2022-12-13 20-17-30

the size of shape is small than the inside, a little

**Comment by himself65** [comment]
I noticed that there still have bugs on the drag&click shape block. But I want to make it in next PR

**Comment by doodlewind** [comment]
> I noticed that there still have bugs on the drag&click shape block. But I want to make it in next PR

Cool, sounds good to merge this first 🙌

**Review by doodlewind — APPROVED** [review]

**Pull request merged**